### PR TITLE
Fix genesis block re-entry problem

### LINKED ISSUE #1949: Genesis block re-entry problem still exist
Checking the current round does not work as it's always value of 0 when node starts, we will need to check the LFB. So if LFB > 0, then we don't need to generate genesis block, otherwise, we need to check the latest round, current round may not work at the moment, need to check the code to see if we can get something from persistent storage.

### PR DESCRIPTION
**Pull request by din-mukhammed**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/sharder/sharder/sharder.go [hunk 1]
```diff
@@ -224,9 +224,9 @@ func main() {
 	go sc.StartLFMBWorker(ctx)
 
 	setupBlockStorageProvider(mConf, workdir)
-	if sc.GetCurrentRound() == 0 {
-		sc.SetupGenesisBlock(viper.GetString("server_chain.genesis_block.id"),
-			magicBlock, initStates)
+	latestFinalizedBlock := sc.GetLatestFinalizedBlock()
+	if latestFinalizedBlock == nil || latestFinalizedBlock.Round == 0 {
+		sc.SetupGenesisBlock(viper.GetString("server_chain.genesis_block.id"), magicBlock, initStates)
```
Comment (peterlimg) [inline code/go/0chain.net/sharder/sharder/sharder.go:229]: LatestFinalizedBlock() might work for sharder(need to confirm), but we also need to check the miners, whose LatestFinalizedBlock() seems always nil when on starting.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/sharder/sharder/sharder.go (hunk 1 above)
Comment (din-mukhammed) [reply code/go/0chain.net/sharder/sharder/sharder.go:229]: Added LFB check for miners also

Comment (peterlimg) [reply code/go/0chain.net/sharder/sharder/sharder.go:229]: @din-mukhammed LFB is also always 0 on start for miners, it's actually not easy to fix. Anyway, I'll fix it in another PR for fixing chain stuck and close this one.

**Comment by peterlimg** [comment]
Closed as https://github.com/0chain/0chain/pull/2039#discussion_r1092710726

**Pull request closed without merging**